edbdeploy/data/terraform/aws-rds-aurora/environments/aws/aws.tf

```diff
@@ -0,0 +1,212 @@
+  }
+
+  connection {
+    private_key = file(var.ssh_pub_key)
+  }
+}
+
+resource "aws_instance" "pem_server" {
+  count = var.pem_server["count"]
+
+  ami = var.aws_ami_id
+
+  instance_type          = var.pem_server["instance_type"]
+  key_name               = aws_key_pair.key_pair.id
+  subnet_id              = element(tolist(data.aws_subnet_ids.selected.ids), count.index)
+  vpc_security_group_ids = [var.custom_security_group_id]
+
+  root_block_device {
+    delete_on_termination = "true"
+    volume_size           = var.pem_server["volume"]["size"]
+    volume_type           = var.pem_server["volume"]["type"]
+    iops                  = var.pem_server["volume"]["type"] == "io2" ? var.pem_server["volume"]["iops"] : var.pem_server["volume"]["type"] == "io1" ? var.pem_server["volume"]["iops"] : null
+  }
+
+  tags = {
+    Name       = format("%s-%s%s", var.cluster_name, "pemserver", count.index + 1)
+    Created_By = var.created_by
+  }
+
+  connection {
+    private_key = file(var.ssh_pub_key)
+  }
+}
+
+resource "aws_instance" "barman_server" {
+  count = var.barman_server["count"]
+
+  ami = var.aws_ami_id
+
+  instance_type          = var.barman_server["instance_type"]
+  key_name               = aws_key_pair.key_pair.id
+  subnet_id              = element(tolist(data.aws_subnet_ids.selected.ids), count.index)
+  vpc_security_group_ids = [var.custom_security_group_id]
+
+  root_block_device {
+    delete_on_termination = "true"
+    volume_size           = var.barman_server["volume"]["size"]
+    volume_type           = var.barman_server["volume"]["type"]
+    iops                  = var.barman_server["volume"]["type"] == "io2" ? var.barman_server["volume"]["iops"] : var.barman_server["volume"]["type"] == "io1" ? var.barman_server["volume"]["iops"] : null
+  }
+
+  tags = {
+    Name       = format("%s-%s%s", var.cluster_name, "barman", count.index + 1)
+    Created_By = var.created_by
+  }
+
+  connection {
+    private_key = file(var.ssh_pub_key)
+  }
+}
+
+resource "aws_instance" "pooler_server" {
+  count = var.pooler_server["count"]
+
+  ami = var.aws_ami_id
+
+  instance_type          = var.pooler_server["instance_type"]
+  key_name               = aws_key_pair.key_pair.id
+  subnet_id              = element(tolist(data.aws_subnet_ids.selected.ids), count.index)
+  vpc_security_group_ids = [var.custom_security_group_id]
+
+  root_block_device {
+    delete_on_termination = "true"
+    volume_size           = var.pooler_server["volume"]["size"]
+    volume_type           = var.pooler_server["volume"]["type"]
+    iops                  = var.pooler_server["volume"]["type"] == "io2" ? var.pooler_server["volume"]["iops"] : var.pooler_server["volume"]["type"] == "io1" ? var.pooler_server["volume"]["iops"] : null
+  }
+
+  tags = {
+    Name       = format("%s-%s%s", var.cluster_name, "pooler", count.index + 1)
+    Created_By = var.created_by
+  }
+
+  connection {
+    private_key = file(var.ssh_pub_key)
+  }
+}
+
```

Comment:
Same question here, why do we want to provision barman and pooler in this context?

Reply:
Yes, it could make sense to use connection pooler in front of RDS/Aurora instances. But for the moment, we don't have any reference architecture mixing connection pooler and DBaaS.

@vibhorkumar123 what do you think?

Reply:
+1. However, in the future, we can work on supporting that with RDS/Aurora.
Also, I think we should develop architecture codes for RDS/Aurora as we have for Postgres.
For example:
EDB-RDS-1 -- without pooler
EDB-Aurora-1 -- without pooler
EDB-RDS-2: With pooler
etc...